src/cryptography/hazmat/backends/openssl/rsa.py

```diff
@@ -74,22 +87,32 @@ def _enc_dec_rsa(backend, key, data, padding):
     return _enc_dec_rsa_pkey_ctx(backend, key, data, padding_enum, padding)
 
 
-def _enc_dec_rsa_pkey_ctx(backend, key, data, padding_enum, padding):
+def _enc_dec_rsa_pkey_ctx(
+    backend,
+    key: typing.Union["_RSAPrivateKey", "_RSAPublicKey"],
+    data: bytes,
+    padding_enum: int,
+    padding: AsymmetricPadding,
+) -> bytes:
     if isinstance(key, _RSAPublicKey):
         init = backend._lib.EVP_PKEY_encrypt_init
         crypt = backend._lib.EVP_PKEY_encrypt
     else:
         init = backend._lib.EVP_PKEY_decrypt_init
         crypt = backend._lib.EVP_PKEY_decrypt
 
-    pkey_ctx = backend._lib.EVP_PKEY_CTX_new(key._evp_pkey, backend._ffi.NULL)
+    pkey_ctx = backend._lib.EVP_PKEY_CTX_new(
+        key._evp_pkey, backend._ffi.NULL  # type: ignore[union-attr]
```

Comment:
Now that this is typed using the implementation, not the interface, I claim that this ignore shouldn't be required.